google_analytics/notebooks/dynamic_aggregation.ipynb

```diff
@@ -33,9 +43,69 @@
     "from aggregation import *"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "deletable": true,
+    "editable": true
+   },
+   "source": [
+    "## Managing a huge file\n",
+    "\n",
+    "Below is the new version of `load`, where processing takes place in chunks. After all chunks have been processed, they are concatenated to a single file. Since many columns are either dropped or aggregated, the resulting dataframe fits in RAM."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false,
+    "deletable": true,
+    "editable": true
+   },
+   "outputs": [],
+   "source": [
+    "# Only run these the first time - after it you can just load the reduced_datasets.\n",
+    "reduce_df(\"../data/train_v2.csv\", output=\"../data/reduced_train.csv\", nrows=None, chunksize=20000)\n",
+    "reduce_df(\"../data/test_v2.csv\", output=\"../data/reduced_test.csv\", nrows=None, chunksize=20000)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false,
+    "deletable": true,
+    "editable": true
+   },
+   "outputs": [],
+   "source": [
+    "print(\"Let's widen the train dataset.\")\n",
```

Comment:
Now we aggregate based on the given train and test periods. However, if we want to change the periods of train_x, train_y, test_x, and test_y, maybe it is good to use the function 'create_train_test' before doing the aggregation. This function is splitting the train and test as given by Kaggle into train and test sets based on a given period.  Furthermore, in this function visitors that visited once and never bought during the training period are dropped. If I am right, this is not done yet. 
To summarize my thoughts a bit, I was thinking about the following steps
1. Load train_v2 and test_v2.
2. Drop unnecessary columns.
3. Respecify train_x, train_y, test_x, and test_y based on given periods. 
4. Drop visitors that visit once and did not buy in x from both x and y.
5. Aggregate the four datasets per visitor (for x: OHE or boolean (for example isUSA) for static and create dynamic features; for y: sum of the revenue)
6. Keep only columns in train_x and test_x that appear in both datasets. 
I am not sure if my thoughts are right, but are this the right steps, do I miss something, or are there maybe unnecessary steps?

Reply:
I was thinking that setting a time frame is now as simple as dropping certain columns (corresponding to the dates outside the time-frame you want). But I might be missing something, let's discuss this in person